File: Java Extras/src/Snippets/TowersOfHanoi.java
package Snippets;

public class TowersOfHanoi {

    public static void main(String[] args) {
        int n = 3; // Number of disks
        towerOfHanoi(n, 'A', 'C', 'B');  // A, B and C are names of rods
    }

    public static void towerOfHanoi(int n, char source, char destination, char helper) {
        if (n == 1) {
            System.out.println("Move disk 1 from rod " + source + " to rod " + destination);
            return;
        }
        towerOfHanoi(n - 1, source, helper, destination);
        System.out.println("Move disk " + n + " from rod " + source + " to rod " + destination);
        towerOfHanoi(n - 1, helper, destination, source);
    }
}

File: Data Structures/src/recursions/TowersOfHanoi.kt
package recursions

const val N = 3 // Number of disks

fun main() = towerOfHanoi(N, 'A', 'B', 'C') // A, B and C are names of rods


fun towerOfHanoi(n: Int, source: Char, helper: Char, destination: Char) {
    if (n == 1) {
        println("Move disk 1 from rod $source to rod $destination")
        return
    }
    towerOfHanoi(n - 1, source = source, helper = destination, destination = helper)
    println("Move disk $n from rod $source to rod $destination")
    towerOfHanoi(n - 1, source = helper, helper = source, destination = destination)
}